File: 0020-valid-parentheses/0020-valid-parentheses.java
class Solution {
    public boolean isValid(String s) {
        Stack<Character> st = new Stack<Character>();

        for(char it : s.toCharArray()){
            if(it == '(' || it == '[' || it == '{'){
                st.push(it);
            }
            else{
                char ch = st.pop();
                if(st.isEmpty()){
                    return false;
                }else{
                    if((it == ')' && ch=='(') || (it == ']' && ch=='[') || (it == '}' && ch=='{'))  continue;
                    else return false;
                }
            }
        }
        return st.isEmpty();
    }
}

File: 0020-valid-parentheses/0020-valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        st = []

        for ch in s:
            if ch == '(' or ch == '[' or ch == '{':
                st.append(ch)

            else:
                if not st:
                    return False

                top = st.pop()

                if ch == ')' and top != '(':
                    return False
                elif ch == ']' and top != '[':
                    return False
                elif ch == '}' and top != '{':
                    return False

        return len(st) == 0